File: day13/Python/fibonacci.py
"""
  @author : vishalshirke7
  @date : 07/01/2019
"""


# Fibonacci Series using Dynamic Programming
def fibonacci(n):
    if n <= 1:
        return n
    else:
        if fib_series[n - 1] == 0:
            fib_series[n - 1] = fibonacci(n - 1)

        if fib_series[n - 2] == 0:
            fib_series[n - 2] = fibonacci(n - 2)

        fib_series[n] = fib_series[n - 2] + fib_series[n - 1]
    return fib_series[n]


n = int(input())
fib_series = [0, 1]
while len(fib_series) < n + 1:
    fib_series.append(0)
print(fibonacci(n))
if n == 0:
    print(0)
else:
    print(", ".join(map(str, fib_series)))

File: day13/C#/fibonacci.cs
/**
 * @author: sagunjaiswal
 * @date: 20/09/2020
 */
using System;  
namespace FibonacciSeries  
{  
      class Program  
      {  
            public static int FibonacciSeries(int n)  
            {  
                  if (n == 0) 
                    return 1; 
                  if (n == 1)
                    return 1;
                  return FibonacciSeries(n - 1) + FibonacciSeries(n - 2);  
            }  
            public static void Main(string[] args)  
            {  
                  Console.Write("input : ");  
                  int length = Convert.ToInt32(Console.ReadLine()); 
                  Console.Write("output : \n"); 
                  Console.Write("The {0}th term of fibonacci series is : {1} \n",length,FibonacciSeries(length-1));  
                  Console.Write("The sequence is : ");
                  for (int i = 0; i < length; i++)  
                  {  
                        Console.Write("{0} ", FibonacciSeries(i));  
                  }  
                
                  Console.ReadKey();  
            }  
      }  
} 